python/cudf/cudf/utils/cudautils.py

```diff
@@ -105,7 +106,7 @@ def gpu_mark_found_int(arr, val, out, not_found):
 def gpu_mark_found_float(arr, val, out, not_found):
     i = cuda.grid(1)
     if i < arr.size:
-        if check_equals_float(arr[i], val):
+        if check_equals_float(arr[i], float(val) if NUMBA_LE_0501 else val):
```

Comment:
Instead of checking for the numba version can we just always do the cast here? Is there any real performance penalty? I figure this kernel is just memory bandwidth bound anyway so this would presumably be "free".

Reply:
Yes, what you said is correct. Here is a benchmark which I have done with variations:

```python
s = cudf.Series([1, 2, 3, 4, 5]*100000, index=[5.43, 6.34, 7.34, 8.0, 9.1]* 100000)

Without any cast:

In [9]: %timeit s.loc[8]
2.53 ms ± 23.6 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

Always cast:

In [4]: %timeit s.loc[8]
3.08 ms ± 14.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

Conditional cast:

In [4]: %timeit s.loc[8]
3.08 ms ± 26.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

```

There is a slight improvement if there is no cast, So added a `TODO` which should be addressed as part of numba upgrade.